Source/Libraries/FaultData/DataResources/SCADADataResource.cs

```diff
@@ -78,79 +96,35 @@ public bool DidBreakerOpen(Asset line, DateTime approximateTime)
             DateTime utcClearingTime = TimeZoneInfo.ConvertTimeToUtc(approximateTime, SystemSettings.XDATimeZoneInfo);
             DateTime localClearingTime = utcClearingTime.ToLocalTime();
 
-            TimeSpan queryTolerance = EDNASettings.QueryToleranceSpan;
+            TimeSpan queryTolerance = SCADASettings.QueryToleranceSpan;
             DateTime startTime = localClearingTime - queryTolerance;
             DateTime endTime = localClearingTime + queryTolerance;
 
-            double breakerOpenValue = EDNASettings.BreakerOpenValue;
+            double breakerOpenValue = SCADASettings.BreakerOpenValue;
 
-            foreach (string point in points)
-            {
-                var previousPoint = new
-                {
-                    Value = default(double),
-                    Time = default(DateTime),
-                    Status = default(string),
-                    Valid = false
-                };
-
-                int[] expectedResults =
-                {
-                    (int)eDNAHistoryReturnStatus.END_OF_HISTORY,
-                    (int)eDNAHistoryReturnStatus.NO_HISTORY_FOR_TIME
-                };
-
-                int result = History.DnaGetHistRaw(point, startTime, endTime, out uint key);
-
-                while (result == 0)
-                {
-                    result = History.DnaGetNextHist(key, out double value, out DateTime time, out string status);
-
-                    if (result == 0)
-                    {
-                        // Verify that the data point represents a change
-                        // from closed to open within the queried time range
-                        bool trip =
-                            previousPoint.Valid &&
-                            previousPoint.Value != breakerOpenValue &&
-                            value == breakerOpenValue &&
-                            time >= startTime &&
-                            time <= endTime;
-
-                        if (trip)
-                            return true;
-
-                        previousPoint = new
-                        {
-                            Value = value,
-                            Time = time,
-                            Status = status,
-                            Valid = true
-                        };
-                    }
-                }
-
-                // Assume that unexpected return status indicates an error
-                // and therefore the analysis results should be trusted
-                if (!expectedResults.Contains(result))
-                {
-                    Log.Debug($"Unexpected eDNA return code: {result}");
-                    return true;
-                }
-            }
-
-            return false;
+            return SCADAHistorianResource.DidBreakerOpen(points, startTime, endTime, breakerOpenValue);
```

Comment:
Forgot to implement a null check.

Reply:
It wasn't as much of an issue as I thought it was because this line of code wouldn't be reached unless SCADA points are configured for the breaker. Most likely, that would imply the SCADA historian is also configured. However, I added the null check anyway to avoid potential issues related to temporarily disabling the SCADA historian, in case that comes up for whatever reason.